ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/admin/ledger_pruning_service.proto

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.digitalasset.ledger.api.v1.admin;
+
+import "google/protobuf/timestamp.proto";
+
+option java_outer_classname = "LedgerPruningServiceOuterClass";
+option java_package = "com.digitalasset.ledger.api.v1.admin";
+option csharp_namespace = "Com.DigitalAsset.Ledger.Api.V1.Admin";
+
+// Status: experimental interface, will change before it is deemed production
+// ready
+
+// Prunes/truncates the "oldest" ledger transactions from the ledger (specifically the Ledger Api Server) by removing
+// a portion of the ledger that does not affect the set of allowed, future commands). This enables:
+// 1. keeping the "inactive" portion of the ledger to a manageable size and
+// 2. removing inactive state to honor right-to-forget guarantees.
+service LedgerPruningService {
+
+  // Prune the ledger specifying the point in time or offset-space before which ledger transactions should be removed.
+  // Returns an error if the ledger api server is not configured to support pruning.
+  rpc PruneByOffset (PruneByOffsetRequest) returns (PruneResponse);
+
+  // Prune the ledger specifying the point in time or offset-space before which ledger transactions should be removed.
+  // Returns an error if the ledger api server is not configured to support pruning or is configured with a static
+  // time (such that time is does not differentiate specific pruning points).
+  rpc PruneByTime (PruneByTimeRequest) returns (PruneResponse);
+}
+
+message PruneByOffsetRequest {
+  google.protobuf.Timestamp prune_before = 1;
```

Comment:
What does `before` mean? Is it inclusive? The name seems to imply an exclusive behavior, which makes sense: it is easier to manage on the `byTime` alternative and it would be nice to have a symmetric meaning on the `byOffset` alternative. Whichever way we want to adopt, it's probably better to document the semantics for clarity's sake.